### PR TITLE
fix: deepcopy lr in RexLR scheduler.

### PR DESCRIPTION
**Pull request by nyxkrage**
This fixes a problem where when the lr is a scalar tensor, the base_lrs in the get_lr function end up being references to the current learning rate, rather than the correct initial learning rate.

See also related pytorch PR https://github.com/pytorch/pytorch/pull/127190/

I discovered this when the during a run with the learning_rate set to 5e-5, the reported learning rate after the warmup ended up being 4.7e-7 instead.
I narrowed it down to being due to the lr being a tensor, and thus found the above linked pytorch PR, and after applying the change to the rexlr constructor, the learning rate correctly peaks at the given learning rate.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of learning rate values in the scheduler to prevent issues when the learning rate is a tensor.

### CODE REVIEW

**Review by winglian — APPROVED** [review]
great find! thanks!
